diff --git a/node_modules/@vot.js/core/dist/client.js b/node_modules/@vot.js/core/dist/client.js
index d439a91..013f649 100644
--- a/node_modules/@vot.js/core/dist/client.js
+++ b/node_modules/@vot.js/core/dist/client.js
@@ -6,6 +6,7 @@ import { fetchWithTimeout, getTimestamp } from "@vot.js/shared/utils/utils";
 import { YandexVOTProtobuf, YandexSessionProtobuf } from "./protobuf.js";
 import { AudioDownloadType, VideoTranslationStatus } from "./types/yandex.js";
 import { convertVOT } from "./utils/vot.js";
+import { VOTLocalizedError } from "../../../../src/utils/VOTLocalizedError.js";
 export class VOTJSError extends Error {
     data;
     constructor(message, data = undefined) {
@@ -186,7 +187,7 @@ export default class VOTClient extends MinimalClient {
             ...headers,
         });
         if (!res.success) {
-            throw new VOTJSError("Failed to request video translation", res);
+            throw new VOTLocalizedError("requestTranslationFailed");
         }
         const translationData = YandexVOTProtobuf.decodeTranslationResponse(res.data);
         console.log(translationData);
@@ -194,11 +195,11 @@ export default class VOTClient extends MinimalClient {
         const { status, translationId, } = translationData;
         switch (status) {
             case VideoTranslationStatus.FAILED:
-                throw new VOTJSError("Yandex couldn't translate video", translationData);
+                throw translationData?.message ? new VOTJSError("Yandex couldn't translate video", translationData) : new VOTLocalizedError("requestTranslationFailed");
             case VideoTranslationStatus.FINISHED:
             case VideoTranslationStatus.PART_CONTENT:
                 if (!translationData.url) {
-                    throw new VOTJSError("Audio link wasn't received from Yandex response", translationData);
+                    throw new VOTLocalizedError("audioNotReceived");
                 }
                 return {
                     translationId,
@@ -257,7 +258,7 @@ export default class VOTClient extends MinimalClient {
             ...headers,
         });
         if (!res.success) {
-            throw new VOTJSError("Failed to request video translation", res);
+            throw new VOTLocalizedError("requestTranslationFailed");
         }
         const translationData = res.data;
         switch (translationData.status) {
@@ -265,7 +266,7 @@ export default class VOTClient extends MinimalClient {
                 throw new VOTJSError("Yandex couldn't translate video", translationData);
             case "success":
                 if (!translationData.translated_url) {
-                    throw new VOTJSError("Audio link wasn't received from VOT response", translationData);
+                    throw new VOTLocalizedError("audioNotReceived");
                 }
                 return {
                     translationId: String(translationData.id),
@@ -435,7 +436,7 @@ export default class VOTClient extends MinimalClient {
     async translateStream({ videoData, requestLang = this.requestLang, responseLang = this.responseLang, headers = {}, }) {
         const { url } = videoData;
         if (this.isCustomLink(url)) {
-            throw new VOTJSError("Unsupported video URL for getting stream translation");
+            throw new VOTLocalizedError("VOTStreamNotSupportedUrl");
         }
         const session = await this.getSession("video-translation");
         const body = YandexVOTProtobuf.encodeStreamRequest(url, requestLang, responseLang);
