diff --git a/node_modules/vot.js/dist/client.js b/node_modules/vot.js/dist/client.js
index 4857f88..dda51d0 100644
--- a/node_modules/vot.js/dist/client.js
+++ b/node_modules/vot.js/dist/client.js
@@ -8,7 +8,7 @@ import { getVideoData } from "./utils/videoData.js";
 import { convertVOT } from "./utils/vot.js";
 import { StreamInterval } from "./protos/yandex.js";
 import { VOTLocalizedError } from "../../../src/utils/VOTLocalizedError.js";
-import debug from "../../../src/utils/debug.js";
+import debug from "../../../src/utils/debug.ts";
 const { version } = packageInfo;
 class VOTJSError extends Error {
     data;
