diff --git a/node_modules/vot.js/dist/config/sites.js b/node_modules/vot.js/dist/config/sites.js
index c4a4a5c..ab11302 100644
--- a/node_modules/vot.js/dist/config/sites.js
+++ b/node_modules/vot.js/dist/config/sites.js
@@ -219,7 +219,7 @@ export default [
     {
         host: VideoService.dailymotion,
         url: "https://dai.ly/",
-        match: /^geo.dailymotion.com$/,
+        match: /^geo([\d]+)?.dailymotion.com$/,
         selector: ".player",
     },
     {
