diff --git a/node_modules/vot.js/dist/config/sites.js b/node_modules/vot.js/dist/config/sites.js
index 55b05f1..b907ad4 100644
--- a/node_modules/vot.js/dist/config/sites.js
+++ b/node_modules/vot.js/dist/config/sites.js
@@ -373,6 +373,14 @@ export default [
         needExtraData: true,
         needBypassCSP: true,
     },
+    {
+        host: VideoService.udemy,
+        url: "https://www.udemy.com/",
+        match: /udemy.com$/,
+        selector:
+            'div[data-purpose="curriculum-item-viewer-content"] > section > div > div > div > div:nth-of-type(2)',
+        needExtraData: true,
+    },
     {
         host: VideoService.watchpornto,
         url: "https://watchporn.to/",
diff --git a/node_modules/vot.js/dist/helpers/index.d.ts b/node_modules/vot.js/dist/helpers/index.d.ts
index 68542a3..c371c46 100644
--- a/node_modules/vot.js/dist/helpers/index.d.ts
+++ b/node_modules/vot.js/dist/helpers/index.d.ts
@@ -13,6 +13,7 @@ import OdyseeHelper from "./odysee.js";
 import CoursehunterLikeHelper from "./coursehunterLike.js";
 import TwitchHelper from "./twitch.js";
 import SapHelper from "./sap.js";
+import UdemyHelper from "./udemy.js";
 import LinkedinHelper from "./linkedin.js";
 import VimeoHelper from "./vimeo.js";
 import YandexDiskHelper from "./yandexdisk.js";
@@ -56,6 +57,7 @@ export * as OdyseeHelper from "./odysee.js";
 export * as CoursehunterLikeHelper from "./coursehunterLike.js";
 export * as TwitchHelper from "./twitch.js";
 export * as SapHelper from "./sap.js";
+export * as UdemyHelper from "./udemy.js";
 export * as LinkedinHelper from "./linkedin.js";
 export * as VimeoHelper from "./vimeo.js";
 export * as YandexDiskHelper from "./yandexdisk.js";
@@ -100,6 +102,7 @@ export declare const availableHelpers: {
     coursehunterLike: typeof CoursehunterLikeHelper;
     twitch: typeof TwitchHelper;
     sap: typeof SapHelper;
+    udemy: typeof UdemyHelper;
     linkedin: typeof LinkedinHelper;
     vimeo: typeof VimeoHelper;
     yandexdisk: typeof YandexDiskHelper;
diff --git a/node_modules/vot.js/dist/helpers/index.js b/node_modules/vot.js/dist/helpers/index.js
index 62ca005..e60fa97 100644
--- a/node_modules/vot.js/dist/helpers/index.js
+++ b/node_modules/vot.js/dist/helpers/index.js
@@ -13,6 +13,7 @@ import OdyseeHelper from "./odysee.js";
 import CoursehunterLikeHelper from "./coursehunterLike.js";
 import TwitchHelper from "./twitch.js";
 import SapHelper from "./sap.js";
+import UdemyHelper from "./udemy.js";
 import LinkedinHelper from "./linkedin.js";
 import VimeoHelper from "./vimeo.js";
 import YandexDiskHelper from "./yandexdisk.js";
@@ -56,6 +57,7 @@ export * as OdyseeHelper from "./odysee.js";
 export * as CoursehunterLikeHelper from "./coursehunterLike.js";
 export * as TwitchHelper from "./twitch.js";
 export * as SapHelper from "./sap.js";
+export * as UdemyHelper from "./udemy.js";
 export * as LinkedinHelper from "./linkedin.js";
 export * as VimeoHelper from "./vimeo.js";
 export * as YandexDiskHelper from "./yandexdisk.js";
@@ -100,6 +102,7 @@ export const availableHelpers = {
     [VideoService.coursehunterLike]: CoursehunterLikeHelper,
     [VideoService.twitch]: TwitchHelper,
     [VideoService.sap]: SapHelper,
+    [VideoService.udemy]: UdemyHelper,
     [VideoService.linkedin]: LinkedinHelper,
     [VideoService.vimeo]: VimeoHelper,
     [VideoService.yandexdisk]: YandexDiskHelper,
diff --git a/node_modules/vot.js/dist/helpers/udemy.d.ts b/node_modules/vot.js/dist/helpers/udemy.d.ts
new file mode 100644
index 0000000..11e0918
--- /dev/null
+++ b/node_modules/vot.js/dist/helpers/udemy.d.ts
@@ -0,0 +1,21 @@
+import * as Udemy from "../types/helpers/udemy.js";
+
+export declare class UdemyHelper {
+  getModuleData(): {
+      courseId?: number;
+  }
+  getLectureId(): null | string;
+  getLectureData(courseId: number|string, lectureId: number|string): Promise<Udemy.Lecture>;
+  getCourseLang(courseId: number|string): Promise<Udemy.Course>;
+  findVideoUrl(sources: Udemy.MediaSource[]): string|null;
+  findSubtitleUrl(captions: Udemy.Caption[], detectedLanguage: RequestLang): string|null;
+  getVideoData(videoId: string): Promise<undefined | {
+      url: string;
+      duration: number;
+      title: string;
+      description: string;
+      translationHelp: TranslationHelp[] | null;
+      detectedLanguage?: RequestLang;
+  }>;
+  getVideoId(url: URL): Promise<string | undefined>;
+}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/helpers/udemy.js b/node_modules/vot.js/dist/helpers/udemy.js
new file mode 100644
index 0000000..1861f2e
--- /dev/null
+++ b/node_modules/vot.js/dist/helpers/udemy.js
@@ -0,0 +1,126 @@
+import { BaseHelper } from "./base.js";
+import debug from "../../../../src/utils/debug.ts";
+import { normalizeLang } from "../utils/utils.js";
+import { localizationProvider } from "../../../../src/localization/localizationProvider.js";
+import { availableLangs } from "../consts.js";
+import sites from "../config/sites.js";
+import { VideoService } from "../types/yandex.js";
+
+export default class UdemyHelper extends BaseHelper {
+  API_ORIGIN = "https://www.udemy.com/api-2.0";
+
+  getModuleData() {
+      const moduleArgs = document.querySelector(
+          ".ud-app-loader[data-module-id='course-taking']",
+      )?.dataset?.moduleArgs;
+      if (!moduleArgs) {
+          console.error(localizationProvider.get("udemyModuleArgsNotFound"));
+          return {};
+      }
+      return JSON.parse(moduleArgs);
+  }
+
+  getLectureId() {
+      return /learn\/lecture\/([^/]+)/.exec(window.location.pathname)?.[1];
+  }
+
+  async getLectureData(courseId, lectureId) {
+      const res = await this.fetch(
+          `${this.API_ORIGIN}/users/me/subscribed-courses/${courseId}/lectures/${lectureId}/?` +
+          new URLSearchParams({
+              "fields[lecture]": "title,description,asset",
+              "fields[asset]": "length,media_sources,captions",
+          })
+      );
+      return await res.json();
+  }
+
+  async getCourseLang(courseId) {
+      const res = await this.fetch(
+          `${this.API_ORIGIN}/users/me/subscribed-courses/${courseId}?` +
+          new URLSearchParams({
+              "fields[course]": "locale",
+          })
+      );
+      return await res.json();
+  }
+
+  findVideoUrl(sources) {
+      return sources?.find((src) => src.type === "video/mp4")?.src;
+  }
+
+  findSubtitleUrl(captions, detectedLanguage) {
+      let subtitle = captions?.find(
+          (caption) => normalizeLang(caption.locale_id) === detectedLanguage,
+      );
+
+      if (!subtitle) {
+          subtitle = captions?.find(
+              (caption) => normalizeLang(caption.locale_id) === "en",
+          ) ?? captions?.[0];
+      }
+
+      return subtitle?.url;
+  }
+
+  async getVideoData(videoId) {
+      const { courseId } = this.getModuleData();
+      if (!courseId) {
+          return undefined;
+      }
+
+      const lectureId = this.getLectureId();
+      debug.log(`[Udemy] courseId: ${courseId}, lectureId: ${lectureId}`)
+      if (!lectureId) {
+          return undefined;
+      }
+
+      const { title, description, asset } = await this.getLectureData(courseId, lectureId);
+      const { length: duration, media_sources, captions } = asset;
+
+      const videoUrl = this.findVideoUrl(media_sources);
+      if (!videoUrl) {
+          console.log("Failed to find .mp4 video file in media_sources", media_sources);
+          return undefined;
+      }
+
+      const courseLangData = await this.getCourseLang(courseId);
+      let { locale: { locale: courseLang } } = courseLangData;
+      courseLang = courseLang ? normalizeLang(courseLang) : "en";
+      if (!availableLangs.includes(courseLang)) {
+          courseLang = "en";
+      }
+
+      const subtitleUrl = this.findSubtitleUrl(captions, courseLang);
+      if (!subtitleUrl) {
+          console.log("Failed to find subtitle file in captions", captions)
+      }
+
+      return {
+          ...subtitleUrl ? {
+              url: this.service?.url + videoId,
+              translationHelp: [
+                  {
+                      target: "subtitles_file_url",
+                      targetUrl: subtitleUrl,
+                  },
+                  {
+                      target: "video_file_url",
+                      targetUrl: videoUrl,
+                  },
+              ],
+              detectedLanguage: courseLang,
+          } : {
+              url: videoUrl,
+              translationHelp: null,
+          },
+          duration,
+          title,
+          description,
+      };
+  }
+
+  async getVideoId(url) {
+      return url.pathname.slice(1);
+  }
+}
diff --git a/node_modules/vot.js/dist/types/helpers/udemy.d.ts b/node_modules/vot.js/dist/types/helpers/udemy.d.ts
new file mode 100644
index 0000000..c097789
--- /dev/null
+++ b/node_modules/vot.js/dist/types/helpers/udemy.d.ts
@@ -0,0 +1,51 @@
+import { ISODate } from "../utils.js";
+
+export type Caption = {
+  _class: "caption";
+  asset_id: number;
+  created: ISODate;
+  file_name: string;
+  id: number;
+  locale_id: string;
+  source: string;
+  status: number;
+  title: string;
+  url: string;
+  video_label: string;
+};
+
+export type MediaSource = {
+  label: string;
+  src: string;
+  type: string;
+};
+
+export type Asset = {
+  _class: "asset";
+  captions: Caption[];
+  id: number;
+  length: number;
+  media_sources: MediaSource[];
+}
+
+export type Lecture = {
+  _class: "lecture";
+  asset: Asset[];
+  description: string;
+  id: number;
+  title: string;
+}
+
+export type Locale = {
+  _class: "locale";
+  english_title: string;
+  locale: string;
+  simple_english_title: string;
+  title: string;
+}
+
+export type Course = {
+  _class: "course";
+  id: number;
+  locale: Locale;
+}
diff --git a/node_modules/vot.js/dist/types/helpers/udemy.js b/node_modules/vot.js/dist/types/helpers/udemy.js
new file mode 100644
index 0000000..693da49
--- /dev/null
+++ b/node_modules/vot.js/dist/types/helpers/udemy.js
@@ -0,0 +1 @@
+export {}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/types/yandex.d.ts b/node_modules/vot.js/dist/types/yandex.d.ts
index 5306eb8..f5db258 100644
--- a/node_modules/vot.js/dist/types/yandex.d.ts
+++ b/node_modules/vot.js/dist/types/yandex.d.ts
@@ -66,6 +66,7 @@ export declare enum VideoService {
     odysee = "odysee",
     coursehunterLike = "coursehunterLike",
     sap = "sap",
+    udemy = "udemy",
     watchpornto = "watchpornto",
     linkedin = "linkedin",
     ricktube = "ricktube",
diff --git a/node_modules/vot.js/dist/types/yandex.js b/node_modules/vot.js/dist/types/yandex.js
index 46ca4a6..0487f7e 100644
--- a/node_modules/vot.js/dist/types/yandex.js
+++ b/node_modules/vot.js/dist/types/yandex.js
@@ -50,6 +50,7 @@ export var VideoService;
     VideoService["odysee"] = "odysee";
     VideoService["coursehunterLike"] = "coursehunterLike";
     VideoService["sap"] = "sap";
+    VideoService["udemy"] = "udemy";
     VideoService["watchpornto"] = "watchpornto";
     VideoService["linkedin"] = "linkedin";
     VideoService["ricktube"] = "ricktube";