diff --git a/node_modules/vot.js/dist/utils/videoData.js b/node_modules/vot.js/dist/utils/videoData.js
index 2532b39..7c1acae 100644
--- a/node_modules/vot.js/dist/utils/videoData.js
+++ b/node_modules/vot.js/dist/utils/videoData.js
@@ -151,6 +151,8 @@ export async function getVideoID(service, video) {
             return /(video|embed)\/([^/]+)/.exec(url.pathname)?.[2];
         case VideoService.coursera:
             return /learn\/([^/]+)\/lecture\/([^/]+)/.exec(url.pathname)?.[0]; // <-- COURSE PASSING (IF YOU LOGINED TO COURSERA)
+        case VideoService.udemy:
+            return url.pathname.slice(1);
         case VideoService.eporner:
             return /video-([^/]+)\/([^/]+)/.exec(url.pathname)?.[0];
         case VideoService.peertube:
@@ -225,7 +227,7 @@ export async function getVideoID(service, video) {
 export async function getVideoData(service, video) {
     const videoId = await getVideoID(service, video);
     if (!videoId) {
-        throw new VideoDataError(`Entered unsupported link: "${url}"`);
+        throw new VideoDataError(`Entered unsupported link: "${service.host}"`);
     }
     if (service.host === VideoService.peertube) {
         service.url = new URL(url).origin;
