diff --git a/node_modules/vot.js/dist/protos/yandex.d.ts b/node_modules/vot.js/dist/protos/yandex.d.ts
index faf4237..88d81fc 100644
--- a/node_modules/vot.js/dist/protos/yandex.d.ts
+++ b/node_modules/vot.js/dist/protos/yandex.d.ts
@@ -1,4 +1,4 @@
-import _m0 from "protobufjs/minimal.js";
+export type _m0 = typeof protobuf;
 export declare const protobufPackage = "";
 export declare enum StreamInterval {
     NO_CONNECTION = 0,
diff --git a/node_modules/vot.js/dist/protos/yandex.js b/node_modules/vot.js/dist/protos/yandex.js
index ee9c885..8020466 100644
--- a/node_modules/vot.js/dist/protos/yandex.js
+++ b/node_modules/vot.js/dist/protos/yandex.js
@@ -1,4 +1,4 @@
-import _m0 from "protobufjs/minimal.js";
+const _m0 = protobuf;
 export const protobufPackage = "";
 export var StreamInterval;
 (function (StreamInterval) {
diff --git a/node_modules/vot.js/dist/secure.js b/node_modules/vot.js/dist/secure.js
index 46df1a1..cf8dc02 100644
--- a/node_modules/vot.js/dist/secure.js
+++ b/node_modules/vot.js/dist/secure.js
@@ -1,4 +1,3 @@
-import crypto from "node:crypto";
 import config from "./config/config.js";
 import Logger from "./utils/logger.js";
 const utf8Encoder = new TextEncoder();