diff --git a/node_modules/vot.js/dist/utils/videoData.d.ts b/node_modules/vot.js/dist/utils/videoData.d.ts
index 1627ef4..a0a4de3 100644
--- a/node_modules/vot.js/dist/utils/videoData.d.ts
+++ b/node_modules/vot.js/dist/utils/videoData.d.ts
@@ -1,6 +1,6 @@
 import { ServiceConf } from "../types/yandex.js";
 import { VideoData } from "../types/client.js";
 export declare function getService(videoUrl: string): false | ServiceConf | undefined;
-export declare function getVideoID(service: ServiceConf, videoURL: string): Promise<string | null | undefined>;
+export declare function getVideoID(service: ServiceConf, video: HTMLVideoElement): Promise<string | null | undefined>;
 export declare function getVideoData(url: string): Promise<VideoData>;
 //# sourceMappingURL=videoData.d.ts.map
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/utils/videoData.js b/node_modules/vot.js/dist/utils/videoData.js
index 6884d08..319e1c8 100644
--- a/node_modules/vot.js/dist/utils/videoData.js
+++ b/node_modules/vot.js/dist/utils/videoData.js
@@ -8,18 +8,11 @@ class VideoDataError extends Error {
         this.message = message;
     }
 }
-export function getService(videoUrl) {
-    if (/(file:\/\/|(http(s)?:\/\/)(127\.0\.0\.1|localhost))/.exec(videoUrl))
-        return false;
-    let enteredURL;
-    try {
-        enteredURL = new URL(videoUrl);
-    }
-    catch (e) {
-        console.error(`Invalid URL: ${videoUrl}. Have you forgotten https?`);
-        return false;
-    }
-    const hostname = enteredURL.hostname;
+export function getService() {
+    if (/(http(s)?:\/\/)(127\.0\.0\.1|localhost)/.exec(window.location.href))
+        return [];
+    const hostname = window.location.hostname;
+    const enteredURL = new URL(window.location);
     const isMathes = (match) => {
         if (match instanceof RegExp) {
             return match.test(hostname);
@@ -32,14 +25,14 @@ export function getService(videoUrl) {
         }
         return false;
     };
-    return sites.find((e) => {
+    return sites.filter((e) => {
         return ((Array.isArray(e.match) ? e.match.some(isMathes) : isMathes(e.match)) &&
             e.host &&
             e.url);
     });
 }
-export async function getVideoID(service, videoURL) {
-    const url = new URL(videoURL);
+export async function getVideoID(service, video) {
+    const url = new URL(window.location.href);
     const serviceHost = service.host;
     if (Object.getOwnPropertyDescriptor(VideoHelper, serviceHost)) {
         return await VideoHelper[serviceHost].getVideoId(url);
@@ -71,6 +64,10 @@ export async function getVideoID(service, videoURL) {
             else if (paramOID && paramID) {
                 return `video-${Math.abs(parseInt(paramOID))}_${paramID}`;
             }
+            const videoBox = video.parentElement?.closest(".video_box_wrap");
+            if (videoBox) {
+                return videoBox.id.replace("video_box_wrap", "video");
+            }
             return null;
         }
         case VideoService.nine_gag:
@@ -115,9 +112,16 @@ export async function getVideoID(service, videoURL) {
         case VideoService.peertube:
             return /\/w\/([^/]+)/.exec(url.pathname)?.[0];
         case VideoService.dailymotion: {
-            return url.hostname === "dai.ly"
-                ? url.pathname.slice(1)
-                : /video\/([^/]+)/.exec(url.pathname)?.[1];
+            // we work in the context of the player
+            // geo.dailymotion.com
+            const plainPlayerConfig = Array.from(
+                document.querySelectorAll("*"),
+            ).filter((s) => s.innerHTML.trim().includes(".m3u8"));
+            let videoUrl = plainPlayerConfig?.[1]?.lastChild?.src;
+            if (!videoUrl) {
+                return undefined
+            }
+            return /\/video\/(\w+)\.m3u8/.exec(videoUrl)?.[1];
         }
         case VideoService.trovo: {
             const vid = url.searchParams.get("vid");
@@ -136,7 +140,7 @@ export async function getVideoID(service, videoURL) {
             return /\/video\/(\d+)/.exec(url.pathname)?.[1];
         }
         case VideoService.googledrive:
-            return /\/file\/d\/([^/]+)/.exec(url.pathname)?.[1];
+            return url.searchParams.get("docid");
         case VideoService.newgrounds:
             return /([^/]+)\/(view)\/([^/]+)/.exec(url.pathname)?.[0];
         case VideoService.egghead:
@@ -149,17 +153,13 @@ export async function getVideoID(service, videoURL) {
             return undefined;
     }
 }
-export async function getVideoData(url) {
-    const service = getService(url);
-    if (!service) {
-        throw new VideoDataError(`URL: "${url}" is unknown service`);
-    }
-    const videoId = await getVideoID(service, url);
+export async function getVideoData(service, video) {
+    const videoId = await getVideoID(service, video);
     if (!videoId) {
-        throw new VideoDataError(`Entered unsupported link: "${url}"`);
+        throw new VideoDataError(`Entered unsupported link: "${service.host}"`);
     }
     if (service.host === VideoService.peertube) {
-        service.url = new URL(url).origin;
+        service.url = window.location.origin;
     }
     if (service.rawResult) {
         return {
