diff --git a/node_modules/vot.js/dist/utils/helper.js b/node_modules/vot.js/dist/utils/helper.js
index b61b6bc..f8eded2 100644
--- a/node_modules/vot.js/dist/utils/helper.js
+++ b/node_modules/vot.js/dist/utils/helper.js
@@ -18,8 +18,7 @@ export class MailRuHelper {
     API_ORIGIN = "https://my.mail.ru/";
     async getExtraVideoId(pathname) {
         try {
-            const res = await fetchWithTimeout(`${this.API_ORIGIN}${pathname}`);
-            const content = await res.text();
+            const content = document.querySelector(".sp-video__page-config")?.innerText;
             return /"itemId":\s?"([^"]+)"/.exec(content)?.[1];
         }
         catch (err) {
