diff --git a/node_modules/vot.js/dist/config/sites.js b/node_modules/vot.js/dist/config/sites.js
index 3e05f90..3f42407 100644
--- a/node_modules/vot.js/dist/config/sites.js
+++ b/node_modules/vot.js/dist/config/sites.js
@@ -37,6 +37,7 @@ export default [
         host: VideoService.poketube,
         url: "https://youtu.be/",
         match: sitesPoketube,
+        selector: ".video-player-container",
     },
     {
         additionalData: "mobile",
diff --git a/node_modules/vot.js/dist/utils/videoData.js b/node_modules/vot.js/dist/utils/videoData.js
index cade01f..5e0a547 100644
--- a/node_modules/vot.js/dist/utils/videoData.js
+++ b/node_modules/vot.js/dist/utils/videoData.js
@@ -261,7 +261,7 @@ export async function getVideoData(service, video) {
         throw new VideoDataError(`Entered unsupported link: "${service.host}"`);
     }
     if (service.host === VideoService.peertube) {
-        service.url = new URL(url).origin;
+        service.url = window.location.origin;
     }
     if (service.rawResult) {
         return {
