diff --git a/node_modules/vot.js/dist/client.js b/node_modules/vot.js/dist/client.js
index bfd2a72..e8d5cad 100644
--- a/node_modules/vot.js/dist/client.js
+++ b/node_modules/vot.js/dist/client.js
@@ -8,6 +8,7 @@ import { getVideoData } from "./utils/videoData.js";
 import { convertVOT } from "./utils/vot.js";
 import { StreamInterval } from "./protos/yandex.js";
 import { VOTLocalizedError } from "../../../src/utils/VOTLocalizedError.js";
+import debug from "../../../src/utils/debug.js";
 const { version } = packageInfo;
 class VOTJSError extends Error {
     data;
@@ -192,6 +193,7 @@ export default class VOTClient {
             throw new VOTLocalizedError("requestTranslationFailed", res);
         }
         const translationData = res.data;
+        debug.log("requestTranslation", translationData);
         switch (translationData.status) {
             case "failed":
                 throw new VOTJSError("Yandex couldn't translate video", translationData);
diff --git a/node_modules/vot.js/dist/utils/videoData.js b/node_modules/vot.js/dist/utils/videoData.js
index 5e0a547..158f35a 100644
--- a/node_modules/vot.js/dist/utils/videoData.js
+++ b/node_modules/vot.js/dist/utils/videoData.js
@@ -210,7 +210,7 @@ export async function getVideoID(service, video) {
             return /\/video\/(\d+)/.exec(url.pathname)?.[1];
         }
         case VideoService.googledrive:
-            return /\/file\/d\/([^/]+)/.exec(url.pathname)?.[1];
+            return url.searchParams.get("docid");
         case VideoService.bannedvideo: {
             return url.searchParams.get("id");
         }
