diff --git a/node_modules/vot.js/dist/config/sites.js b/node_modules/vot.js/dist/config/sites.js
index 9c2d6a0..199665d 100644
--- a/node_modules/vot.js/dist/config/sites.js
+++ b/node_modules/vot.js/dist/config/sites.js
@@ -370,6 +370,14 @@ export default [
             'div[data-purpose="curriculum-item-viewer-content"] > section > div > div > div > div:nth-of-type(2)',
         needExtraData: true,
     },
+    {
+        // ONLY IF YOU LOGINED TO COURSERA /learn/NAME/lecture/XXXX
+        host: VideoService.coursera,
+        url: "https://www.coursera.org/",
+        match: /coursera.org$/,
+        selector: ".vjs-v8",
+        needExtraData: true,
+    },
     {
         host: VideoService.watchpornto,
         url: "https://watchporn.to/",
diff --git a/node_modules/vot.js/dist/helpers/coursera.d.ts b/node_modules/vot.js/dist/helpers/coursera.d.ts
new file mode 100644
index 0000000..fc46e72
--- /dev/null
+++ b/node_modules/vot.js/dist/helpers/coursera.d.ts
@@ -0,0 +1,16 @@
+import * as Coursera from "../types/helpers/coursera.js";
+
+export declare class CourseraHelper {
+  getCourseData(courseId: number|string): Promise<Coursera.Course>;
+  getPlayer(): HTMLDivElement | null;
+  getPlayerData(): Coursera.PlayerData;
+  findVideoUrl(sources: Coursera.Source[]): string|null;
+  findSubtitleUrl(captions: Coursera.Track[], detectedLanguage: RequestLang): string|null;
+  getVideoData(videoId: string): Promise<false | {
+      url: string;
+      duration: number;
+      translationHelp: TranslationHelp[] | null;
+      detectedLanguage?: RequestLang;
+  }>;
+  getVideoId(url: URL): Promise<string | undefined>;
+}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/helpers/coursera.js b/node_modules/vot.js/dist/helpers/coursera.js
new file mode 100644
index 0000000..b59fbc5
--- /dev/null
+++ b/node_modules/vot.js/dist/helpers/coursera.js
@@ -0,0 +1,94 @@
+import { BaseHelper } from "./base.js";
+import { normalizeLang } from "../utils/utils.js";
+import { availableLangs } from "vot.js/consts";
+import sites from "vot.js/sites";
+import { VideoService } from "../types/yandex.js";
+
+export default class CourseraHelper extends BaseHelper {
+    API_ORIGIN = "https://www.coursera.org/api";
+
+    async getCourseData(courseId) {
+        const response = await this.fetch(
+            `${this.API_ORIGIN}/onDemandCourses.v1/${courseId}`,
+        );
+        const resJSON = await response.json();
+        return resJSON?.elements?.[0];
+    }
+
+    getPlayer() {
+        return document.querySelector(".vjs-v8");
+    }
+
+    getPlayerData() {
+        return this.getPlayer()?.player;
+    }
+
+    findVideoUrl(sources) {
+        return sources?.find((src) => src.type === "video/mp4")?.src;
+    }
+
+    findSubtitleUrl(captions, detectedLanguage) {
+        let subtitle = captions?.find(
+            (caption) => normalizeLang(caption.srclang) === detectedLanguage,
+        );
+
+        if (!subtitle) {
+            subtitle = captions?.find(
+                (caption) => normalizeLang(caption.srclang) === "en",
+            ) || captions?.[0];
+        }
+
+        return subtitle?.src;
+    }
+
+    async getVideoData(videoId) {
+        const data = this.getPlayerData();
+
+        const { duration } = data?.cache_ || {};
+        const { courseId, tracks, sources } = data?.options_ || {};
+
+        const videoUrl = this.findVideoUrl(sources);
+        if (!videoUrl) {
+            console.log("Failed to find .mp4 video file in sources", sources);
+            return undefined;
+        }
+
+        const { primaryLanguageCodes } = await this.getCourseData(courseId);
+        let courseLang = primaryLanguageCodes?.[0];
+        courseLang = courseLang ? normalizeLang(courseLang) : "en";
+
+        if (!availableLangs.includes(courseLang)) {
+            courseLang = "en";
+        }
+
+        const subtitleUrl = this.findSubtitleUrl(tracks, courseLang);
+        if (!subtitleUrl) {
+            console.log("Failed to find subtitle file in tracks", tracks)
+        }
+
+        return {
+            ...subtitleUrl ? {
+                url: this.service?.url + videoId,
+                translationHelp: [
+                    {
+                        target: "subtitles_file_url",
+                        targetUrl: subtitleUrl,
+                    },
+                    {
+                        target: "video_file_url",
+                        targetUrl: videoUrl,
+                    },
+                ],
+            } : {
+                url: videoUrl,
+                translationHelp: null,
+            },
+            detectedLanguage: courseLang,
+            duration,
+        };
+    }
+
+    async getVideoId(url) {
+        return /learn\/([^/]+)\/lecture\/([^/]+)/.exec(url.pathname)?.[0]; // <-- COURSE PASSING (IF YOU LOGINED TO COURSERA)
+    }
+}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/helpers/index.d.ts b/node_modules/vot.js/dist/helpers/index.d.ts
index 26077c7..4bb3148 100644
--- a/node_modules/vot.js/dist/helpers/index.d.ts
+++ b/node_modules/vot.js/dist/helpers/index.d.ts
@@ -14,6 +14,7 @@ import CoursehunterLikeHelper from "./coursehunterLike.js";
 import TwitchHelper from "./twitch.js";
 import SapHelper from "./sap.js";
 import UdemyHelper from "./udemy.js";
+import CourseraHelper from "./coursera.js";
 import LinkedinHelper from "./linkedin.js";
 import VimeoHelper from "./vimeo.js";
 import YandexDiskHelper from "./yandexdisk.js";
@@ -36,6 +37,7 @@ export * as CoursehunterLikeHelper from "./coursehunterLike.js";
 export * as TwitchHelper from "./twitch.js";
 export * as SapHelper from "./sap.js";
 export * as UdemyHelper from "./udemy.js";
+export * as CourseraHelper from "./coursera.js";
 export * as LinkedinHelper from "./linkedin.js";
 export * as VimeoHelper from "./vimeo.js";
 export * as YandexDiskHelper from "./yandexdisk.js";
@@ -59,6 +61,7 @@ export declare const availableHelpers: {
     twitch: typeof TwitchHelper;
     sap: typeof SapHelper;
     udemy: typeof UdemyHelper;
+    coursera: typeof CourseraHelper;
     linkedin: typeof LinkedinHelper;
     vimeo: typeof VimeoHelper;
     yandexdisk: typeof YandexDiskHelper;
diff --git a/node_modules/vot.js/dist/helpers/index.js b/node_modules/vot.js/dist/helpers/index.js
index 28a55a5..9c488d4 100644
--- a/node_modules/vot.js/dist/helpers/index.js
+++ b/node_modules/vot.js/dist/helpers/index.js
@@ -14,6 +14,7 @@ import CoursehunterLikeHelper from "./coursehunterLike.js";
 import TwitchHelper from "./twitch.js";
 import SapHelper from "./sap.js";
 import UdemyHelper from "./udemy.js";
+import CourseraHelper from "./coursera.js";
 import LinkedinHelper from "./linkedin.js";
 import VimeoHelper from "./vimeo.js";
 import YandexDiskHelper from "./yandexdisk.js";
@@ -36,6 +37,7 @@ export * as CoursehunterLikeHelper from "./coursehunterLike.js";
 export * as TwitchHelper from "./twitch.js";
 export * as SapHelper from "./sap.js";
 export * as UdemyHelper from "./udemy.js";
+export * as CourseraHelper from "./coursera.js";
 export * as LinkedinHelper from "./linkedin.js";
 export * as VimeoHelper from "./vimeo.js";
 export * as YandexDiskHelper from "./yandexdisk.js";
@@ -59,6 +61,7 @@ export const availableHelpers = {
     [VideoService.twitch]: TwitchHelper,
     [VideoService.sap]: SapHelper,
     [VideoService.udemy]: UdemyHelper,
+    [VideoService.coursera]: CourseraHelper,
     [VideoService.linkedin]: LinkedinHelper,
     [VideoService.vimeo]: VimeoHelper,
     [VideoService.yandexdisk]: YandexDiskHelper,
diff --git a/node_modules/vot.js/dist/types/helpers/coursera.d.ts b/node_modules/vot.js/dist/types/helpers/coursera.d.ts
new file mode 100644
index 0000000..2803fc9
--- /dev/null
+++ b/node_modules/vot.js/dist/types/helpers/coursera.d.ts
@@ -0,0 +1,52 @@
+export type Course = {
+  categoryIds: unknown[];
+  courseStatus: string;
+  description: string;
+  enrollableSiteUserRoles: unknown[];
+  estimatedWorkload: string;
+  id: string;
+  instructorIds: string[];
+  isReal: boolean;
+  isRestrictedMembership: boolean;
+  isSubtitleTranslationEnabled: boolean;
+  isVerificationEnabled: boolean;
+  launchedAt: number;
+  name: string;
+  partnerIds: string[];
+  previewUserIds: unknown[];
+  primaryLanguageCodes: string[];
+  promoPhoto: string;
+  slug: string;
+  subtitleLanguageCodes: string[];
+  verificationEnabledAt: number;
+}
+
+// not fully typed
+export type Cache = {
+  duration: number;
+}
+
+export type Source = {
+  src: string;
+  type: string;
+}
+
+export type Track = {
+  kind: "captions";
+  label: string;
+  src: string;
+  srclang: string;
+}
+
+// not fully typed
+export type Options = {
+  courseId: string;
+  sources: Source[];
+  tracks: Track[]
+}
+
+// not fully typed
+export type PlayerData = {
+  cache_: Cache;
+  options_: Options;
+}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/types/helpers/coursera.js b/node_modules/vot.js/dist/types/helpers/coursera.js
new file mode 100644
index 0000000..693da49
--- /dev/null
+++ b/node_modules/vot.js/dist/types/helpers/coursera.js
@@ -0,0 +1 @@
+export {}
\ No newline at end of file
diff --git a/node_modules/vot.js/dist/types/yandex.d.ts b/node_modules/vot.js/dist/types/yandex.d.ts
index 555b4ac..783def9 100644
--- a/node_modules/vot.js/dist/types/yandex.d.ts
+++ b/node_modules/vot.js/dist/types/yandex.d.ts
@@ -60,6 +60,7 @@ export declare enum VideoService {
     coursehunterLike = "coursehunterLike",
     sap = "sap",
     udemy = "udemy",
+    coursera = "coursera",
     watchpornto = "watchpornto",
     linkedin = "linkedin",
     ricktube = "ricktube",
diff --git a/node_modules/vot.js/dist/types/yandex.js b/node_modules/vot.js/dist/types/yandex.js
index 520989c..963c358 100644
--- a/node_modules/vot.js/dist/types/yandex.js
+++ b/node_modules/vot.js/dist/types/yandex.js
@@ -50,6 +50,7 @@ export var VideoService;
     VideoService["coursehunterLike"] = "coursehunterLike";
     VideoService["sap"] = "sap";
     VideoService["udemy"] = "udemy";
+    VideoService["coursera"] = "coursera";
     VideoService["watchpornto"] = "watchpornto";
     VideoService["linkedin"] = "linkedin";
     VideoService["ricktube"] = "ricktube";
