diff --git a/node_modules/vot.js/dist/config/sites.js b/node_modules/vot.js/dist/config/sites.js
index ab11302..b91a1fb 100644
--- a/node_modules/vot.js/dist/config/sites.js
+++ b/node_modules/vot.js/dist/config/sites.js
@@ -331,12 +331,14 @@ export default [
         host: VideoService.epicgames,
         url: "https://dev.epicgames.com/community/learning/",
         match: /^dev.epicgames.com$/,
+        selector: ".vjs-v7",
         needExtraData: true,
     },
     {
         host: VideoService.nineanimetv,
         url: "https://9animetv.to/watch/",
-        match: /^9animetv.to$/,
+        match: /^rapid-cloud.co$/,
+        selector: ".jw-media",
         needExtraData: true,
     },
     {
diff --git a/node_modules/vot.js/dist/utils/helper.js b/node_modules/vot.js/dist/utils/helper.js
index 5933938..d0c0942 100644
--- a/node_modules/vot.js/dist/utils/helper.js
+++ b/node_modules/vot.js/dist/utils/helper.js
@@ -642,7 +642,7 @@ export class NineAnimetvHelper {
     RAPID_CLOUD_ORIGIN = "https://rapid-cloud.co/ajax/embed-6-v2";
     async getSourceId(episodeId) {
         try {
-            const res = await fetchWithTimeout(`${this.API_ORIGIN}/servers?episodeId=${episodeId}`);
+            const res = await GM_fetch(`${this.API_ORIGIN}/servers?episodeId=${episodeId}`);
             const content = (await res.json());
             if (!content.html) {
                 return false;
@@ -656,7 +656,7 @@ export class NineAnimetvHelper {
     }
     async getPlayerLink(sourceId) {
         try {
-            const res = await fetchWithTimeout(`${this.API_ORIGIN}/sources?id=${sourceId}`);
+            const res = await GM_fetch(`${this.API_ORIGIN}/sources?id=${sourceId}`);
             const content = (await res.json());
             if (!content.link.includes("rapid-cloud.co")) {
                 return false;
@@ -670,7 +670,7 @@ export class NineAnimetvHelper {
     }
     async getRapidCloudData(rapidId) {
         try {
-            const res = await fetchWithTimeout(`${this.RAPID_CLOUD_ORIGIN}/getSources?id=${rapidId}`);
+            const res = await GM_fetch(`${this.RAPID_CLOUD_ORIGIN}/getSources?id=${rapidId}`);
             const content = (await res.json());
             if (content.encrypted) {
                 console.warn("Encrypted RapidCloud data found. Let us know about it", content);
diff --git a/node_modules/vot.js/dist/utils/videoData.js b/node_modules/vot.js/dist/utils/videoData.js
index 188abc1..c5d8b30 100644
--- a/node_modules/vot.js/dist/utils/videoData.js
+++ b/node_modules/vot.js/dist/utils/videoData.js
@@ -255,7 +255,19 @@ export async function getVideoID(service, video) {
             return /\/(\w{4})\/[^/]+$/.exec(url.pathname)?.[1];
         }
         case VideoService.nineanimetv: {
-            return /[^/]+$/.exec(url.href)?.[0];
+            return new Promise(resolve => {
+                const origin = "https://9animetv.to"
+                window.addEventListener("message", (e) => {
+                    if (e.origin !== origin) {
+                        return;
+                    }
+
+                    if (e.data?.startsWith("getVideoId:")) {
+                        resolve(e.data.replace("getVideoId:"))
+                    }
+                })
+                window.top.postMessage("getVideoId", origin);
+            })
         }
         default:
             return undefined;
